cmd/k8s-operator/sts.go

```diff
@@ -294,6 +294,7 @@ func (a *tailscaleSTSReconciler) reconcileHeadlessService(ctx context.Context, l
 			Selector: map[string]string{
 				"app": sts.ParentResourceUID,
 			},
+			IPFamilyPolicy: ptr.To(corev1.IPFamilyPolicyPreferDualStack),
```

Comment:
Question: 

>  instead only did so when switching the cluster DNS to Cloud DNS

Did we actually need to set the IP family to `PreferDualStack` for it to work with CloudDNS?

Reply:
Yeah, I just re-tested without this change to be sure, but it defaults to `SingleStack` if left blank: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.30/#servicespec-v1-core. So in that case even with CloudDNS we still only get an A record, and no AAAA record.